modules/shared/src/main/scala/io/constellationnetwork/schema/swap.scala

```diff
@@ -158,14 +159,35 @@ object swap {
   @newtype
   case class SpendTransactionFee(value: PosLong)
 
+  @derive(decoder, encoder, order, ordering, show)
+  sealed trait SpendTransaction extends SwapTransaction
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class SpendTransactionReference(hash: Hash)
+
+  object SpendTransactionReference {
+    def of[F[_]: Async](spendTransaction: SpendTransaction)(implicit hasher: Hasher[F]): F[SpendTransactionReference] =
+      hasher.hash(spendTransaction).map(SpendTransactionReference(_))
+  }
   @derive(decoder, encoder, order, show)
-  case class SpendTransaction(
+  case class CreatedSpendTransaction(
     fee: SpendTransactionFee,
-    lastValidOrdinal: SnapshotOrdinal,
+    lastValidEpochProgress: EpochProgress,
     allowSpendRef: Hash,
     currency: Option[CurrencyId],
     amount: SwapAmount
-  )
+  ) extends SpendTransaction
+
+  @derive(decoder, encoder, order, show)
+  case class PendingSpendTransaction(
```

Comment:
What is a difference between Created and Pending? I thought we just need Pending and Completed

Reply:
👍🏼